flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyMessageDecoderDelegate.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.netty;
+
+import org.apache.flink.shaded.netty4.io.netty.buffer.ByteBuf;
+import org.apache.flink.shaded.netty4.io.netty.buffer.ByteBufAllocator;
+
+import javax.annotation.Nullable;
+
+/**
+ * Parsers for specified netty messages.
+ */
+public interface NettyMessageDecoderDelegate {
+
+	/**
+	 * The result of message parsing with the provided data.
+	 */
+	class ParseResult {
+		private final static ParseResult NOT_FINISHED = new ParseResult(false, null);
+
+		static ParseResult notFinished() {
+			return NOT_FINISHED;
+		}
+
+		static ParseResult finishedWith(@Nullable NettyMessage message) {
+			return new ParseResult(true, message);
+		}
+
+		final boolean finished;
+
+		@Nullable
+		final NettyMessage message;
+
+		private ParseResult(boolean finished, NettyMessage message) {
+			this.finished = finished;
+			this.message = message;
+		}
+	}
+
+	/**
+	 * Notifies the underlying channel become active.
+	 *
+	 * @param alloc The netty buffer allocator.
+	 */
+	void onChannelActive(ByteBufAllocator alloc);
```

Comment:
It seems more general for giving `ChannelHandlerContext` instead of `ByteBufAllocator` in interface.
But I have no strong preference for it, you can consider a bit.

Reply:
Have modified accordingly.